pkg/util/cmapichecker/cmapichecker.go

```diff
@@ -0,0 +1,115 @@
+/*
+Copyright 2021 The cert-manager Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmapichecker
+
+import (
+	"context"
+
+	errors "github.com/pkg/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	// Use v1alpha2 API to ensure that the API server has also connected to the
+	// cert-manager conversion webhook.
+	// TODO(wallrj): Only change this when the old deprecated APIs are removed,
+	// at which point the conversion webhook may be removed anyway.
+	cmapi "github.com/jetstack/cert-manager/pkg/apis/certmanager/v1alpha2"
+	cmmeta "github.com/jetstack/cert-manager/pkg/apis/meta/v1"
+)
+
+// Interface is used to check that the cert-manager CRDs have been installed and are usable.
+type Interface interface {
+	Check(context.Context) error
+}
+
+type cmapiChecker struct {
+	// The client controller-runtime client.New function fails if can't reach
+	// the API server, so we load it lazily, to avoid breaking integration tests
+	// which rely on being able to start the webhook server before the API
+	// server.
```

Comment:
This was a hack when I wrote it in the startup probe branch, but I don't think it is necessary here....although perhaps it could play a part if the resulting error message is presented to the user as `Unable to reach Kubernetes API server`...I'm not sure.

Reply:
I think it is useful. It makes the tool easier to use I think. I set `unable to connect to the Kubernetes API server` as error message.